lib/Backend/LowerMDShared.cpp

```diff
@@ -8683,24 +8687,21 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
             //     if (shouldCheckNegZero) {
             //         CMP roundedFloat, 0
             //         JA $setZero
-            //         JE $negZeroTest
-            //         J $bailout
+            //         JP $skipRoundSd (NaN)
+            //       $negZeroTest [Helper]:
+            //         JNE $bailoutLabel
+            //         isNegZero(src)
+            //         JE $bailoutLabel
+            //         J $skipRoundSd
             //     } else {
             //         CMP roundedFloat, 0
-            //         JL $bailout
-            //         J $setZero
-            //     }
+            //         JNE $setZero
```

Comment:
Yup, now that I realize it won't be NaN here I'm changing this as well... but why do you think we need to bailout here? Couldn't whole block just be `J $setZero`?